lib/utils/blockio.js

```diff
@@ -0,0 +1,276 @@
+/*!
+ * blockio.js - blockio object for bcoin
+ * Copyright (c) 2017, Christopher Jeffrey (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+const assert = require('assert');
+const fs = require('../utils/fs');
+const path = require('path');
+const util = require('../utils/util');
+const crc = require('../utils/crc');
+const Network = require('../protocol/network');
+const StaticWriter = require('../utils/staticwriter');
+const FileLocation = require('../utils/fileloc');
+
+/**
+ * File representation of a Block
+ * Serialized to a file region as
+ * <4 bytes network magic bytes>
+ * + <4 bytes raw block length>
+ * + <n bytes raw block>
+ * + <4 bytes CRC32 checksum>
+ * @alias module:utils.FileBlock
+ * @constructor
+ * @param {Buffer} block
+ * @param {String} network
+ */
+
+function FileBlock(block, network) {
+  if (!(this instanceof FileBlock))
+    return new FileBlock(block, network);
+
+  this.block = block;
+  this.network = network;
+}
+
+/**
+ * Serialize the FileBlock.
+ * @returns {Buffer}
+ */
+
+FileBlock.prototype.toRaw = function toRaw() {
+  const bw = new StaticWriter(this.block.length + 12);
+
+  bw.writeU32(this.network.magic);
+  bw.writeU32(this.block.length);
+  bw.writeBytes(this.block);
```

Comment:
It would be better to write the block to disk directly and build the CRC incrementally. This prevents a reallocation of 1mb+ just for a write.